airflow/providers/docker/operators/docker.py

```diff
@@ -269,14 +269,17 @@ def _run_image(self) -> Optional[str]:
             # duplicated conditional logic because of expensive operation
             ret = None
             if self.do_xcom_push:
-                ret = self.cli.logs(container=self.container['Id']) if self.xcom_all else line.encode('utf-8')
+                if self.xcom_all:
+                    ret = self.cli.logs(container=self.container['Id'])
+                else:
+                    ret = self.cli.logs(container=self.container['Id'], tail=1).strip()
 
             if self.auto_remove:
                 self.cli.remove_container(self.container['Id'])
 
             return ret
 
-    def execute(self, context) -> Optional[str]:
+    def execute(self, context) -> Optional[bytes]:
```

Comment:
should we perhaps decode utf-8 (assuming that's what comes out of logs, so that we return a string type? sincere question... 🤔

Reply:
coincidence someone is doing exactly that: https://github.com/apache/airflow/pull/13536/files

Reply:
Yes, and I think this would make more sense, but again I wanted to avoid a breaking change...

I would actually be happy with this other PR being merged, as it now includes the fix in my PR.